test/promql/tests/test_persistentvolumeissues.yaml

```diff
@@ -0,0 +1,137 @@
+
+rule_files:
+  - ../extracted-rules.yaml
+
+tests:
+  # Alerted cases:
+  - interval: 1m
+    input_series:
+
+      # Persistent volume is in the Pending state so it will be alerted.
+      - series: 'kube_persistentvolume_status_phase{persistentvolume="pv-1", namespace="ns-1", phase="Pending"}'
+        values: '1x4'
+      - series: 'kube_persistentvolume_status_phase{persistentvolume="pv-1", namespace="ns-1", phase="Available"}'
+        values: '0x4'
+      - series: 'kube_persistentvolume_status_phase{persistentvolume="pv-1", namespace="ns-1", phase="Bound"}'
+        values: '0x4'
+      - series: 'kube_persistentvolume_status_phase{persistentvolume="pv-1", namespace="ns-1", phase="Released"}'
+        values: '0x4'
+      - series: 'kube_persistentvolume_status_phase{persistentvolume="pv-1", namespace="ns-1", phase="Failed"}'
+        values: '0x4'
+
+    alert_rule_test:
+      - eval_time: 5m
+        alertname: PersistentVolumeIssues
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: ns-1
+              persistentvolume: pv-1
+              phase: Pending
+            exp_annotations:
+              summary: "Persistent Volume pv-1 is in Pending phase for more than 5 minutes"
+              description: "Persistent Volume pv-1 in namespace ns-1 is in Pending phase for more than 5 minutes"
+
+  - interval: 1m
+    input_series:
+      # Persistent volume is in the Failed state so it will be alerted.
+      - series: 'kube_persistentvolume_status_phase{persistentvolume="pv-2", namespace="ns-2", phase="Pending"}'
+        values: '0x4'
+      - series: 'kube_persistentvolume_status_phase{persistentvolume="pv-2", namespace="ns-2", phase="Available"}'
+        values: '0x4'
+      - series: 'kube_persistentvolume_status_phase{persistentvolume="pv-2", namespace="ns-2", phase="Bound"}'
+        values: '0x4'
+      - series: 'kube_persistentvolume_status_phase{persistentvolume="pv-2", namespace="ns-2", phase="Released"}'
+        values: '0x4'
+      - series: 'kube_persistentvolume_status_phase{persistentvolume="pv-2", namespace="ns-2", phase="Failed"}'
+        values: '1x4'
+
+    alert_rule_test:
+      - eval_time: 5m
+        alertname: PersistentVolumeIssues
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: ns-2
+              persistentvolume: pv-2
+              phase: Failed
+            exp_annotations:
+              summary: "Persistent Volume pv-2 is in Failed phase for more than 5 minutes"
+              description: "Persistent Volume pv-2 in namespace ns-2 is in Failed phase for more than 5 minutes"
+
+  # Not Alerted cases:
+  - interval: 1m
+    input_series:
+
+      # Persistent volume is in the Pending state for less than 5 mins so it will not be alerted.
+      - series: 'kube_persistentvolume_status_phase{persistentvolume="pv-3", namespace="ns-3", phase="Pending"}'
+        values: '1x3 0'
```

Comment:
this is similar to the other thread. We have minute 5 with all phases with zero, which I don't think is a valid combination.

Reply:
I removed this test case.